docs/shared/concepts/sync-generators.md

```diff
@@ -0,0 +1,46 @@
+# Sync Generators
+
+In Nx 19.7, you can use sync generators to update code before a task is run or before code is sent to CI. Sync generators are designed to perform maintenance work that is required for a particular task or for CI.
+
+Sync Generator Examples:
+
+- Ensure code is formatted in a specific way before CI is run
```

Comment:
I don't think this is an applicable example, we have `nx format` and making that pluggable would be the preferred way to achieve this IMO (because we have internal utils like `formatFiles` which are coupled to format and won't take sync generators into account in any way)